### PR TITLE
Revive project Part 1: VS 2022 direct support + Header only scan support

### PR DESCRIPTION
**Pull request by Agrael1**
IWYU support:
1. Header file scan support
2. Commentary option (turn on and off commentaries from IWYU)
3. VS 2022 along with VS2019 direct support, updated SDK

Second stage will include: moving to community asynchronous SDK and IWYU migration

### CODE REVIEW

File: IncludeToolbox/IncludeToolbox.csproj [hunk 1]
```diff
@@ -113,25 +114,28 @@
...
     <Reference Include="WindowsBase" />
   </ItemGroup>
   <ItemGroup>
-    <PackageReference Include="Microsoft.VisualStudio.SDK" Version="15.9.3" ExcludeAssets="runtime" />
-    <PackageReference Include="Microsoft.VSSDK.BuildTools" Version="16.0.2268" />
+    <PackageReference Include="Microsoft.VisualStudio.SDK" Version="16.10.31321.278" ExcludeAssets="runtime">
+      <IncludeAssets>compile; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+    <PackageReference Include="Microsoft.VisualStudio.VCProjectEngine">
+      <Version>17.2.32505.173</Version>
```
Comment (Wumpf) [inline IncludeToolbox/IncludeToolbox.csproj:133]: does this cause issues with supporting older visual studio versions?

File: IncludeToolbox/IncludeWhatYouUse/IWYU.cs [hunk 2]
```diff
@@ -249,41 +250,8 @@ static public async Task<string> RunIncludeWhatYouUse(string fullFileName, EnvDT
                 clangOptionList.Add("-w");
                 // ... despite of that "invalid token paste" comes through a lot. Disable it.
                 clangOptionList.Add("-Wno-invalid-token-paste");
-                // Assume C++14
-                clangOptionList.Add("-std=c++14");
                 // MSVC specific. See https://clang.llvm.org/docs/UsersManual.html#microsoft-extensions
                 clangOptionList.Add("-fms-compatibility -fms-extensions -fdelayed-template-parsing");
-                clangOptionList.Add($"-fmsc-version={VSUtils.GetMSCVerString()}");
```
Comment (Wumpf) [inline IncludeToolbox/IncludeWhatYouUse/IWYU.cs:256]: The rest makes sense but why was this bit removed?

File: IncludeToolbox/IncludeWhatYouUse/IWYU.cs [hunk 3]
```diff
@@ -249,41 +250,8 @@ static public async Task<string> RunIncludeWhatYouUse(string fullFileName, EnvDT
...
-                    switch (VSUtils.VCUtils.GetLinkerSetting_TargetMachine(project))
-                    {
-                        // Most targets give an error of this form:
-                        // "error: unknown target CPU 'x86'"
-                        // It seems iwyu is only really fine with x86-64
-
-                        /*case VCProjectUtils.Base.TargetMachineType.X86:
-                            clangOptions.Add("-march=x86");
-                            break;*/
-                        case VCHelper.TargetMachineType.AMD64:
```
Comment (Wumpf) [inline IncludeToolbox/IncludeWhatYouUse/IWYU.cs:269]: not specifying a target machine will cause it to use the default machine for clang which is rather problematic. But what was there was worse/as bad ofc...

[comment not actionable]

File: IncludeToolbox/Options/IncludeWhatYouUseOptionsPage.cs [hunk 4]
```diff
@@ -59,7 +59,13 @@ public void AddMappingFiles(IEnumerable<string> mappingFilesPaths)
         [Category("iwyu options")]
         [DisplayName("No Default Mappings")]
         [Description("Do not add iwyu's default mappings. (--no_default_mappings)")]
-        public bool NoDefaultMappings { get; set; } = false;
+        public bool NoDefaultMappings { get; set; } = false; 
+        
+        [Category("iwyu options")]
+        [DisplayName("No Default Mappings")]
```
Comment (Wumpf) [inline IncludeToolbox/Options/IncludeWhatYouUseOptionsPage.cs]: copy paste error in settings display name & description

**Review by Wumpf — CHANGES_REQUESTED** [review]

**Comment by Wumpf** [comment]
nice stuff in there, thanks :)

File: IncludeToolbox/IncludeToolbox.csproj (hunk 1 above)
Comment (Agrael1) [reply IncludeToolbox/IncludeToolbox.csproj:133]: This handled the problems with compilation. It requested VCProjectEngine, so I installed NuPkg. After step 2, support for versions will be easier with Community SDK.

File: IncludeToolbox/IncludeWhatYouUse/IWYU.cs (hunk 2 above)
Comment (Agrael1) [reply IncludeToolbox/IncludeWhatYouUse/IWYU.cs:256]: unimportant, for the latest built version of IWYU, which I use, this command does not do anything, also, clang has cl support commands, so I utilize them in my project (exempli gratia /std:c++ version query).

File: IncludeToolbox/IncludeWhatYouUse/IWYU.cs (hunk 3 above)
Comment (Agrael1) [reply IncludeToolbox/IncludeWhatYouUse/IWYU.cs:269]: Will be solved later with query options.

File: IncludeToolbox/Options/IncludeWhatYouUseOptionsPage.cs (hunk 4 above)
Comment (Agrael1) [reply IncludeToolbox/Options/IncludeWhatYouUseOptionsPage.cs]: Oof, my bad

Comment (Agrael1) [reply IncludeToolbox/Options/IncludeWhatYouUseOptionsPage.cs]: fixed

**Comment by Agrael1** [comment]
So the feature requested is newly built IWYU with the latest version of clang. The best way is to set up pipeline with periodical checks for an update.

**Review by Wumpf — APPROVED** [review]

**Pull request merged**